src/ray/core_worker/task_event_buffer.cc

```diff
@@ -147,6 +141,15 @@ Status TaskEventBufferImpl::Start(bool auto_flush) {
     } else {
       RAY_LOG(INFO) << "Current task event io thread's nice = " << new_nice;
     }
+#endif
+
+#ifndef _WIN32
+    // Block SIGINT and SIGTERM so they will be handled by the main thread.
+    sigset_t mask;
+    sigemptyset(&mask);
+    sigaddset(&mask, SIGINT);
+    sigaddset(&mask, SIGTERM);
+    pthread_sigmask(SIG_BLOCK, &mask, NULL);
```

Comment:
It's probably a bad idea to delay blocking signals right? What happens if a signal races with `nice` or the other code above?

Reply:
I don't understand this codepath enough to say what risk this is, but seems that we should keep the order as it was.

Reply:
Sure - I coulkd keep the original sequence.